File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def dfs(v):
    visited[v] = True
    print(v, end=' ')
    for i in arr[v]:
        if not visited[i]:
            dfs(i)

def bfs(v):
    queue = deque([v])
    visited[v] = True
    while queue:
        x = queue.popleft()
        print(x, end=' ')
        for i in arr[x]:
            if not visited[i]:
                visited[i] = True
                queue.append(i)


n, m, v = map(int, input().split())

arr = [[] for _ in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split())
    arr[a].append(b)
    arr[b].append(a)
for i in arr:
    i.sort()

visited = [False]*(n+1)
dfs(v)
print()

visited = [False]*(n+1)
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	
	static ArrayList<Integer>[] arr;
	static int n, m, v;
	static boolean[] check;
	static StringBuilder sb;
	
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());
        
        arr = new ArrayList[n+1];
        for(int i = 0; i<arr.length; i++) {
        	arr[i] = new ArrayList<>();
        }
        
        for(int i = 0; i<m; i++) {
        	st = new StringTokenizer(br.readLine());
        	int a = Integer.parseInt(st.nextToken());
        	int b = Integer.parseInt(st.nextToken());
        	arr[a].add(b);
        	arr[b].add(a);
        }
        for(int i = 1; i<arr.length; i++) {
        	Collections.sort(arr[i]);
        }
        check = new boolean[n+1];
        sb = new StringBuilder();
        dfs(v);
        System.out.println(sb);
        check = new boolean[n+1];
        sb = new StringBuilder();
        bfs(v);
        System.out.println(sb);
        
    }
    public static void dfs(int start) {
    	check[start] = true;
    	sb.append(start).append(" ");
    	
    	for(int v: arr[start]) {
    		if(!check[v]) {
    			dfs(v);
    		}
    	}
    }
    public static void bfs(int start) {
    	Queue<Integer> q = new LinkedList<>();
    	q.add(start);
    	check[start] = true;
    	sb.append(start).append(" ");
    	while(!q.isEmpty()) {
    		int now = q.poll();
    		for(int v : arr[now]) {
    			if(!check[v]) {
    				check[v] = true;
    				sb.append(v).append(" ");
    				q.add(v);
    			}
    		}
    	}
    }
    
}
